sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -166,7 +165,10 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
       `IdentityClient: refreshing access token with client ID: ${clientId}, scopes: ${scopes} started`
     );
 
-    const { span, updatedOptions } = createSpan("IdentityClient-refreshAccessToken", options);
+    const { span, updatedOptions } = tracingClient.startSpan(
```

Comment:
I was wondering if it's possible to use `tracingClient.withSpan` here instead? You can still try...catch and rethrow as normal you'd just not have to worry about setting the span status and the active context.

Reply:
That is a problem for this section:

```ts
    } catch (err) {              
      if (                                               
        err.name === AuthenticationErrorName &&    
        err.errorResponse.error === "interaction_required"    
      ) {                                                     
        // It's likely that the refresh token has expired, so    
        // return null so that the credential implementation will    
        // initiate the authentication flow again.    
        logger.info(`IdentityClient: interaction required for client ID: ${clientId}`);    
        span.setStatus({    
          status: "error",    
          error: err,    
        });    
    
        return null;
```

Is there a way to use the span within `withSpan`?

Reply:
Yeah there totally is! The second argument to the callback is a `span`... so something like:

```ts
tracingClient.withSpan("", options, async (updatedOptions, span) => {
    // do something with span
})
```

Although I should add that in this case we don't need to set the status on the span, `withSpan` will do it for you normally. In this case it won't since you are not rethrowing the error (which is totally fine) but one thing we're trying to move away from is setting a status of failure for caught / expected exceptions if that makes sense?

Reply:
oh ok ok! I see. Ok I’ll do the change

Reply:
Done!